kernel/inductive.ml

```diff
@@ -1288,14 +1288,14 @@ let drop_uniform_parameters nuniformparams bodies =
   in
   Array.mapi (fun i -> aux i 0) bodies
 
-let filter_fix_stack_domain nr decrarg stack nuniformparams =
+let filter_fix_stack_domain ?evars nr decrarg stack nuniformparams =
   let rec aux i nuniformparams stack =
     match stack with
     | [] -> []
     | a :: stack ->
       let uniform, nuniformparams = if nuniformparams = 0 then false, 0 else true, nuniformparams -1 in
       let a =
-        if uniform || Int.equal i decrarg then a
+        if uniform || Int.equal i decrarg then SArg (stack_element_specif ?evars a)
```

Comment:
Is it possible to have an SClosure when the argument is the structural one? If not maybe we should assert here.

Reply:
Yes ? That would be the issue this fixes, SClosure may happen outside of the fixpoint but must be reduced to SArg inside of it

Reply:
I assumed from the repro case that this code path was triggered with the uniform argument, not the `i = decrarg` case.

Reply:
There's no uniform argument in the proof of False ? Note that you can replace argument `t` with `tt` and it accepts it just the same
```coq
Fixpoint error (t : unit) {struct t} : False :=
  (fix rec (n : nat) {struct n} :=
    match n with 0 => error | S n => rec n end)
    1 tt.
```